astropy/coordinates/sky_coordinate.py

```diff
@@ -1359,11 +1360,29 @@ def radial_velocity_correction(self, kind='barycentric', obstime=None,
                              "'{}'".format(kind))
 
         gcrs_p, gcrs_v = location.get_gcrs_posvel(obstime)
-        gtarg = self.transform_to(GCRS(obstime=obstime,
-                                       obsgeoloc=gcrs_p,
-                                       obsgeovel=gcrs_v))
-        targcart = gtarg.represent_as(UnitSphericalRepresentation).to_cartesian()
-        return targcart.dot(v_origin_to_earth + gcrs_v)
+        # transforming to GCRS is not the correct thing to do here, since we don't want to
+        # include aberration (or light deflection)? Instead, only apply parallax if necessary
+        if self.data.__class__ is UnitSphericalRepresentation:
+            targcart = self.represent_as(UnitSphericalRepresentation).to_cartesian()
+        else:
+            # skycoord has distances so apply parallax
+            obs_icrs_cart = get_body_barycentric('earth', obstime) + gcrs_p
```

Comment:
It seems a bit wasteful to re-calculate Earth here, when it was already done for the velocity above.
Probably best to just have `pos_earth, vel_earth = get_body_barycentric('earth', obstime)` early on.

Reply:
good idea